src/redux/checkout/checkout.sagas.js

```diff
@@ -0,0 +1,116 @@
+import { call, put, takeEvery } from 'redux-saga/effects';
+
+import { push } from 'connected-react-router';
+import {
+  setNovaPoshtaCities,
+  setNovaPoshtaWarehouse,
+  setNovaPoshtaStreets,
+  setLoading,
+  setNovaPoshtaPrices
+} from './checkout.actions';
+import {
+  GET_NOVAPOSHTA_CITIES,
+  GET_NOVAPOSHTA_WAREHOUSES,
+  GET_NOVAPOSHTA_STREETS,
+  GET_NOVAPOSHTA_PRICES
+} from './checkout.types';
+import getItems from '../../utils/client';
+import { setError } from '../error/error.actions';
+
+export function* handlePrice({ payload }) {
+  try {
+    const price = yield call(
+      getItems,
+      `query {
+        getNovaPoshtaPrices(data:{
+              cityRecipient: "${payload.cityRecipient}"
+              weight: ${payload.weight}
+              cost: ${payload.cost}
+              seatsAmount: 1
+              serviceType: "${payload.serviceType}"
+                    }){
+          cost
+          assessedCost
+        }
+      }`
+    );
+    yield put(setNovaPoshtaPrices(...price.data.getNovaPoshtaPrices));
+  } catch (e) {
+    yield put(setError({ e }));
+    yield put(push('/error-page'));
+  }
+}
+
+export function* handleStreets({ payload }) {
+  try {
+    yield put(setLoading(true));
+    const streets = yield call(
+      getItems,
+      `query($cityRef: String, $street: String){
+                getNovaPoshtaStreets(cityRef: $cityRef, street: $street){   
+                    description
+                    ref
+                    streetsTypeRef
+                    streetsType
+                    }
+                 }`,
+      { cityRef: payload.ref, street: payload.street }
+    );
+    yield put(setNovaPoshtaStreets(streets.data.getNovaPoshtaStreets));
+    yield put(setLoading(false));
+  } catch (e) {
+    yield put(setError({ e }));
+    yield put(push('/error-page'));
```

Comment:
Move this error handling to separate  `handleError` method

Reply:
fixed